pymc3/model.py

```diff
@@ -844,14 +872,23 @@ def __init__(self, name, data, distribution, model):
         name : str
         distribution : Distribution
         model : Model
+        total_size : scalar Tensor (optional)
+            needed for upscaling logp
         """
         self.name = name
         self.data = {name: as_tensor(data, name, model, distribution)
                      for name, data in data.items()}
 
         self.missing_values = [datum.missing_values for datum in self.data.values()
                                if datum.missing_values is not None]
-        self.logp_elemwiset = distribution.logp(**self.data)
+        logp_elemwiset = distribution.logp(**self.data)
+        if total_size is None:
+            coef = tt.constant(1)
+        else:
+            assert logp_elemwiset.ndim >= 1, ('Variable with scaled density '
```

Comment:
should raise an honest error, and best to raise a custom one (`assert` statements can be turned off at run time)